### PR TITLE
Include representational description to classes in params.py

### PR DESCRIPTION
**Pull request by rkbeatss**
Added dunder methods to classes in order to enhance readability

### CODE REVIEW

File: fastapi/params.py [hunk 1]
```diff
@@ -50,8 +50,8 @@ def __init__(
             regex=regex,
             **extra,
         )
-
-
+
```
Comment (Kludex) [inline fastapi/params.py:53]:
```suggestion

```

File: fastapi/params.py [hunk 2]
```diff
@@ -50,8 +50,8 @@ def __init__(
             regex=regex,
             **extra,
         )
-
-
+    
+
```
Comment (Kludex) [inline fastapi/params.py:54]:
```suggestion

```

File: fastapi/params.py (hunk 1 above)
Comment (Kludex) [reply fastapi/params.py:53]: https://softwareengineering.stackexchange.com/questions/121555/why-is-trailing-whitespace-a-big-deal#:~:text=While%20not%20strictly%20programming%2C%20whitespace,as%20input%20to%20something%20else.

File: fastapi/params.py [hunk 3]
```diff
@@ -316,6 +317,12 @@ def __init__(self, dependency: Callable = None, *, use_cache: bool = True):
         self.dependency = dependency
         self.use_cache = use_cache
 
+    def __repr__(self) -> str:
+        attr = getattr(self.dependency, "__name__",
+                       type(self.dependency).__name__)
+
+        cache = '' if self.use_cache else ',use_cache=False'
```
Comment (Kludex) [inline fastapi/params.py:324]: I don't know how important is it, but why not adding the `use_cache=True` as well?

**Review by Kludex — APPROVED** [review]

**Review by Kludex — APPROVED** [review]

**Comment by victorphoenix3** [comment]
@rkbeatss would be nicer to clean the commit history too!

**Comment by rkbeatss** [comment]
closing this pull request now as it seems @victorphoenix3 updated her [PR](https://github.com/tiangolo/fastapi/pull/1541)

**Comment by tiangolo** [comment]
Thanks for your contributions here @rkbeatss and @victorphoenix3 ! :rocket: 

And thanks @Kludex for the code review :bow: 

About git commit history, don't worry about it, the merge is done with squash so it doesn't really matter.

**Pull request closed without merging**